### DISCUSSION TITLE
Using placeholder to ckickable for using camera or gallery

### DISCUSSION THREAD

**Discussion by ZanOdin** [skydoves/landscapist#90] [Q&A] [answered]
Hello. I want to use GlideImage for load my picture from gallery, also i want to use camera for get some picture. Before that if it's possible, can i put placeholder to my glideimage to click on it for start my ActivityResultContract to get content.
Thank you

**Comment by ZanOdin** [comment]
Now my way is:
GlideImage(
                imageModel = if (imageUri == null) placeholder else imageUri,
                contentScale = ContentScale.Crop,
                requestOptions = {
                    RequestOptions()
                        .override(400, 400)
                        .diskCacheStrategy(DiskCacheStrategy.ALL)
                        .centerCrop()
                },
                placeHolder = ImageBitmap.imageResource(id = placeholder),
                modifier = Modifier
                    .height(200.dp)
                    .width(200.dp)
                    .clickable {
                        launcher.launch("image/*")
                    }
            )

But for stop duplicate code i want to do it more correctly

**Comment by skydoves** [comment, marked as answer]
Hey, @ZanOdin 
Do you have any preferred way to resolve it?

**Reply by ZanOdin** [reply]
Hello, @skydoves 
It seems like it's not bad, but maybe there is an implementation better than this one?

**Reply by skydoves** [reply]
I think it seems the best solution. Because the painter (placeHolder) can't have `clickable`.
